### ISSUE TITLE
Node fails with panic (Gnosis, Mainnet) 

### ISSUE THREAD

**Issue by antdanchenko** [ledgerwatch/erigon#7745]
#### System information

Erigon version: 
v2.45.2

OS & Version:
Linux

Commit hash: 
048fe49e4f379829e92116326ed8fcec3c9c8c9a

Chain/Network: 
Gnosis, Mainnet

#### Backtrace

````
panic: no gaps in tx ids are allowed: block 28387486 does jump from 187929377 to 187929409

goroutine 8577 [running]:
github.com/ledgerwatch/erigon/turbo/snapshotsync.DumpTxs.func3.1({0x7ca3f14694e0?, 0xc016052578?, 0x8?}, {0x7ca3f14694e8?, 0x22fb320?, 0xa?})
	github.com/ledgerwatch/erigon/turbo/snapshotsync/block_snapshots.go:1429 +0x314
github.com/ledgerwatch/erigon-lib/kv/mdbx.(*MdbxTx).ForAmount(0x2cb0370?, {0x247f50d?, 0xfc30872386a9bb45?}, {0xc016052578, 0x8, 0x8}, 0xb, 0xc014c20180)
	github.com/ledgerwatch/erigon-lib@v0.0.0-20230607134918-6106e5abdf0b/kv/mdbx/kv_mdbx.go:1911 +0x1a8
github.com/ledgerwatch/erigon/turbo/snapshotsync.DumpTxs.func3({0x2cb0370, 0xc0221a5da0}, {0x7c78eb4f9738?, 0x8?, 0xc0214bf2b0?}, {0x7c78eb4f9740?, 0x0?, 0xc0214bf278?})
	github.com/ledgerwatch/erigon/turbo/snapshotsync/block_snapshots.go:1426 +0x85c
github.com/ledgerwatch/erigon-lib/kv.BigChunks.func1({0x2cb0370, 0xc0221a5da0})
	github.com/ledgerwatch/erigon-lib@v0.0.0-20230607134918-6106e5abdf0b/kv/helpers.go:70 +0x1eb
github.com/ledgerwatch/erigon-lib/kv/mdbx.(*MdbxKV).View(0xdf8475800?, {0x2c92978?, 0xc000130020?}, 0xc01cf05400)
	github.com/ledgerwatch/erigon-lib@v0.0.0-20230607134918-6106e5abdf0b/kv/mdbx/kv_mdbx.go:650 +0xa6
github.com/ledgerwatch/erigon-lib/kv.BigChunks({0x2c99f78, 0xc0005e04e0}, {0x246e5ff, 0xf}, {0xc016052588, 0x8, 0x8}, 0xc01d6c71e0)
	github.com/ledgerwatch/erigon-lib@v0.0.0-20230607134918-6106e5abdf0b/kv/helpers.go:48 +0x192
github.com/ledgerwatch/erigon/turbo/snapshotsync.DumpTxs({0x2c92940?, 0xc0005b2320}, {0x2c99f78?, 0xc0005e04e0}, 0x1b126b8, 0x1b12aa0, 0xc0214bfa20, 0x1, 0x4, {0x2ca4da8, ...}, ...)
	github.com/ledgerwatch/erigon/turbo/snapshotsync/block_snapshots.go:1382 +0x716
github.com/ledgerwatch/erigon/turbo/snapshotsync.dumpBlocksRange({_, _}, _, _, {_, _}, {_, _}, _, {0x2c99f78, ...}, ...)
	github.com/ledgerwatch/erigon/turbo/snapshotsync/block_snapshots.go:1234 +0xbd1
github.com/ledgerwatch/erigon/turbo/snapshotsync.DumpBlocks({0x2c92940, 0xc0005b2320}, 0x1b126b8, 0x1b12aa0, 0x7a120, {0xc000ac5f00, 0x19}, {0xc000ac5f20, 0x1e}, 0xb337508, ...)
	github.com/ledgerwatch/erigon/turbo/snapshotsync/block_snapshots.go:1164 +0x27a
github.com/ledgerwatch/erigon/turbo/snapshotsync.(*BlockRetire).RetireBlocks(0xc011268c80, {0x2c92940, 0xc0005b2320}, 0x1b126b8, 0x1b12aa0, 0x4)
	github.com/ledgerwatch/erigon/turbo/snapshotsync/block_snapshots.go:1077 +0x3a5
github.com/ledgerwatch/erigon/turbo/snapshotsync.(*BlockRetire).RetireBlocksInBackground.func1()
	github.com/ledgerwatch/erigon/turbo/snapshotsync/block_snapshots.go:1147 +0x105
created by github.com/ledgerwatch/erigon/turbo/snapshotsync.(*BlockRetire).RetireBlocksInBackground
	github.com/ledgerwatch/erigon/turbo/snapshotsync/block_snapshots.go:1139 +0xd3
````

**Comment by AskAlexSharov** [comment]
workaround: --snap.stop
fix: integration stage_headers --reset

**Comment by JP-Ellis** [comment]
I encountered this issue with `v2.46.0`. Is it any use to you to see logs around when it started? Or is this a known issue?

**Comment by AskAlexSharov** [comment]
It’s a bit known issue -  but can’t reproduce yet

**Comment by StiofanZ** [comment]
upgrade to v2.47.0

**Comment by WhymustIhaveaname** [comment]
> workaround: --snap.stop fix: integration stage_headers --reset
Could you explain more about `integration stage_headers --reset`? I met the same problem.

**Comment by AskAlexSharov** [comment]
"integration stage_headers --reset" it deleting all blocks from db, it forcing erigon to re-download them and re-save in db. it allows loose all re-orgs effect. this bug is related to reorgs handling logic.
it's fixed in v2.47.0
